game/scripts/vscripts/items/manta.lua

```diff
@@ -0,0 +1,213 @@
+
+  -- Sound
+  caster:EmitSound("DOTA_Item.Manta.Activate")
+
+  -- The user is invulnerable, hidden and spell immune during the split time.
+  caster:AddNewModifier(caster, self, "modifier_item_manta_splitted", { duration = self:GetSpecialValueFor("invuln_duration") })
+
+end
+
+function item_manta:GetIntrinsicModifierName()
+  return "modifier_generic_bonus"
+end
+
+modifier_item_manta_splitted = class(ModifierBaseClass)
+
+function modifier_item_manta_splitted:CheckState()
+  return {
+    [MODIFIER_STATE_OUT_OF_GAME] = true,
+    [MODIFIER_STATE_INVULNERABLE] = true,
+    [MODIFIER_STATE_NO_HEALTH_BAR] = true
+  }
+end
+
+function modifier_item_manta_splitted:GetEffectName()
+  return "particles/items2_fx/item_manta_phase.vpcf"
+end
+
+function modifier_item_manta_splitted:GetEffectAttachType()
+  return PATTACH_ABSORIGIN_FOLLOW
+end
+
+function modifier_item_manta_splitted:OnCreated()
+end
+
+function modifier_item_manta_splitted:IsDebuff()
+  return false
+end
+
+function modifier_item_manta_splitted:IsPurgeable()
+  return false
+end
+
+function modifier_item_manta_splitted:OnDestroy()
+  local ability = self:GetAbility()
+  local caster = self:GetCaster()
+  local playerID = caster:GetPlayerID()
+  local teamID = caster:GetTeam()
+  local forwardVector = caster:GetForwardVector()
+  local origin = caster:GetAbsOrigin()
+
+  local images_count = ability:GetSpecialValueFor("images_count")
+  local image_outgoing_damage = ability:GetSpecialValueFor("damage_outgoing_melee")
+  local image_incoming_damage = ability:GetSpecialValueFor("damage_incoming_melee")
+
+  if caster:IsRangedAttacker() then
+    image_outgoing_damage = ability:GetSpecialValueFor("damage_outgoing_ranged")
+    image_incoming_damage = ability:GetSpecialValueFor("damage_incoming_ranged")
+  end
+
+
+  if ability.images == nil then
+    ability.images = {}
+  end
+
+  -- Get Caster Position Index
+  local casterIndex = RandomInt(1, images_count)
+
+  -- Place Caster
+  FindClearSpaceForUnit(caster, GetImageLocation(caster:GetAbsOrigin(), casterIndex, true, casterIndex, images_count), true)
+
+  DebugDrawSphere(origin, Vector(255, 0, 0), 255, 360, true, 20)
+
+  for imageIndex = 0,images_count do
+    local image = ability.images[imageIndex]
+
+  -- The formation of the owner and the illusions is always the same. One spawns on the owner's cast location and the others randomly on north, east, south or west side each.
+  -- Though the formation is always the same, the owner and the illusions take a random position in the formation and have all the same facing angle.
+    local position = GetImageLocation(origin, casterIndex, false, imageIndex, images_count)
+
+  -- Recasting this replaces the illusions from the previous cast which are currently under the owner's control.
+    if image ~= nil and not image:IsNull() then
+      if image:IsAlive() then
+        position = image:GetAbsOrigin()
```

Comment:
Wouldn't setting `position` to the old illusion's position cause the new one to spawn there? They're supposed to spawn next to you.